src/keys_manager.rs

```diff
@@ -64,4 +92,14 @@ mod test {
         assert_eq!(secret.passphrase, "hodl");
         assert_eq!(secret.seed.len(), 32);
     }
+
+    #[test]
+    fn test_mnemonic_to_secret() {
```

Comment:
I would like to see a test where we actually test against hard-coded values. So generate a seed on https://iancoleman.io/bip39/ (use a password), and also copy the BIP39 seed from there and check whether we end up with the same result